lib/galaxy/tool_util/data/bundles/models.py

```diff
@@ -0,0 +1,270 @@
+        for data_table_name, column in self._walk_columns():
+            data_table_column_name = column.data_table_name
+            output_ref = column.output_ref
+            if output_ref is not None:
+                if data_table_name not in output_refs:
+                    output_refs[data_table_name] = {}
+                output_refs[data_table_name][data_table_column_name] = output_ref
+        return output_refs
+
+    @property
+    def move_by_data_table_column(self) -> Dict[str, Dict[str, DataTableBundleProcessorDataTableOutputColumnMove]]:
+        by_column: Dict[str, Dict[str, DataTableBundleProcessorDataTableOutputColumnMove]] = {}
+        for data_table_name, column in self._walk_columns():
+            data_table_column_name = column.data_table_name
+            for move in column.moves:
+                if data_table_name not in by_column:
+                    by_column[data_table_name] = {}
+                by_column[data_table_name][data_table_column_name] = move
+
+        return by_column
+
+    @property
+    def value_translation_by_data_table_column(self) -> Dict[str, Dict[str, List[Union[str, Callable]]]]:
+        by_column: Dict[str, Dict[str, List[Union[str, Callable]]]] = {}
+        for data_table_name, column in self._walk_columns():
+            data_table_column_name = column.data_table_name
+            for value_translation_model in column.value_translations:
+                value_translation_str = value_translation_model.value
+                value_translation_type = value_translation_model.type
+                if data_table_name not in by_column:
+                    by_column[data_table_name] = {}
+                if data_table_column_name not in by_column[data_table_name]:
+                    by_column[data_table_name][data_table_column_name] = []
+                value_translation: Union[str, Callable]
+                if value_translation_type == "function":
+                    if value_translation_str in VALUE_TRANSLATION_FUNCTIONS:
+                        value_translation = VALUE_TRANSLATION_FUNCTIONS[value_translation_str]
+                    else:
+                        raise ValueError(f"Unsupported value translation function: '{value_translation}'")
+                else:
+                    assert value_translation_type == DEFAULT_VALUE_TRANSLATION_TYPE, ValueError(
+                        f"Unsupported value translation type: '{value_translation_type}'"
+                    )
+                    value_translation = value_translation_str
+                by_column[data_table_name][data_table_column_name].append(value_translation)
+        return by_column
+
+
+class RepoInfo(BaseModel):
+    tool_shed: str
+    name: str
+    owner: str
+    installed_changeset_revision: str
+
+    class Config:
+        # we use dictionary equality (yuk) so definitely make sure this is okay.
+        extra = Extra.forbid
+
+
+class DataTableBundle(BaseModel):
```

Comment:
Could / should this include a serialized version of the job that generated the bundle ?